### PR TITLE
Move reused WAC structs in utils whatsapp package for DRY codes

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: utils/whatsapp/structs.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+type WACMsgTemplating struct {
+	Template struct {
+		Name string `json:"name" validate:"required"`
+		UUID string `json:"uuid" validate:"required"`
+	} `json:"template" validate:"required,dive"`
+	Namespace string   `json:"namespace"`
+	Variables []string `json:"variables"`
+}
+
+type WACMOMedia struct {
```
Comment (rowanseymour) [inline utils/whatsapp/structs.go:12]: We should call these things what they are called by META in their API - not by how we use them. I also good to put some links in comments here pointing to the API docs

File: handlers/whatsapp/whatsapp.go [hunk 2]
```diff
@@ -319,18 +319,6 @@ func (h *handler) BuildAttachmentRequest(ctx context.Context, b courier.Backend,
...
-	"sending":   courier.MsgStatusWired,
-	"sent":      courier.MsgStatusSent,
-	"delivered": courier.MsgStatusDelivered,
-	"read":      courier.MsgStatusDelivered,
-	"failed":    courier.MsgStatusFailed,
-}
-
-var waIgnoreStatuses = map[string]bool{
-	"deleted": true,
-}
```
Comment (rowanseymour) [inline handlers/whatsapp/whatsapp.go:332]: I'm inclined to say we don't touch this handler and instead rename it to `whatsapp_legacy` and let it die

File: utils/whatsapp/variables.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package whatsapp
```
Comment (rowanseymour) [inline utils/whatsapp/variables.go:1]: feels weird separating things into `structs.go` and `variables.go`... both of these could just go in `api.go`

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: utils/whatsapp/variables.go (hunk 3 above)
Comment (rowanseymour) [reply utils/whatsapp/variables.go:1]: The go linter calls this "stuttering" tho only when it repeats exactly, e.g. `whatsapp.WhatsAppTemplate`

File: utils/whatsapp/structs.go [hunk 4]
```diff
@@ -0,0 +1,273 @@
+package whatsapp
+
+type WACMsgTemplating struct {
```
[the remark this answers is not in the document]
Comment (rowanseymour) [reply utils/whatsapp/structs.go:3]: The go linter calls this "stuttering" tho only when it repeats exactly, e.g. `whatsapp.WhatsAppTemplate`

File: handlers/whatsapp/whatsapp.go (hunk 2 above)
Comment (norkans7) [reply handlers/whatsapp/whatsapp.go:332]: I thought so but someone started changing that a few days ago!!

File: utils/whatsapp/variables.go (hunk 3 above)
Comment (norkans7) [reply utils/whatsapp/variables.go:1]: ok

File: utils/whatsapp/structs.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (norkans7) [reply utils/whatsapp/structs.go:3]: Ok

File: handlers/whatsapp/whatsapp.go (hunk 2 above)
Comment (rowanseymour) [reply handlers/whatsapp/whatsapp.go:332]: Haha fair point yeah I should have left it - when you're finished with this pr let's rename this to `whatsapp_legacy`... and maybe also `facebook` to `facebook_legacy` and `facebookapp` to... `meta` ??

File: utils/whatsapp/api.go [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+	Template struct {
+		Name string `json:"name" validate:"required"`
+		UUID string `json:"uuid" validate:"required"`
+	} `json:"template" validate:"required,dive"`
+	Namespace string   `json:"namespace"`
+	Variables []string `json:"variables"`
+}
+
+// API docs https://developers.facebook.com/docs/whatsapp/cloud-api/reference/media#example-2
+type MOMedia struct {
```
Comment (norkans7) [inline utils/whatsapp/api.go:27]: `MOMedia` is a little different from `MTMedia`

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**